### PR TITLE
[Feature] add sys.fe_memory_usage table

### PR DESCRIPTION
**Pull request by Astralidea**
Why I'm doing:
Although we have log monitoring, for some environments, users may not have a way to directly obtain log information and view it, so we expose a metadata table to check the current memory statistics.

What I'm doing:

Fixes #issue

```
+--------------+------------------------------+---------------------+------------------+------------------------------------------------------------+
| module_name  | class_name                   | current_consumption | peak_consumption | counter_info                                               |
+--------------+------------------------------+---------------------+------------------+------------------------------------------------------------+
| Delete       | DeleteMgr                    |                 416 |              416 | {"DeleteInfo":0}                                           |
| Load         | LoadMgr                      |            12187384 |         12187384 | {"LoadJob":0}                                              |
| Load         | RoutineLoadMgr               |                 592 |              592 | {"RoutineLoad":0}                                          |
| Load         | StreamLoadMgr                |                 376 |              376 | {"StreamLoad":0}                                           |
| Task         | TaskManager                  |                  80 |               80 | {"Task":0}                                                 |
| Task         | TaskRunManager               |            12187384 |         12187384 | {"PendingTaskRun":0,"RunningTaskRun":0,"HistoryTaskRun":0} |
| Report       | ReportHandler                |            12188992 |         12188992 | {"PendingTask":0,"ReportQueue":0}                          |
| Backup       | BackupHandler                |            12183864 |         14562256 | {"BackupOrRestoreJob":0}                                   |
| Coordinator  | QeProcessorImpl              |                 240 |              240 | {"QueryInfo":0}                                            |
| Profile      | ProfileManager               |                 112 |              112 | {"Profile":0,"LoadProfile":0}                              |
| Transaction  | GlobalTransactionMgr         |            12183864 |         12183864 | {"Transaction":0}                                          |
| Dict         | CacheDictManager             |            12188768 |         12188768 | {"ColumnDict":0}                                           |
| Export       | ExportMgr                    |                 248 |              248 | {"ExportJob":0}                                            |
| Tablet       | TabletInvertedIndex          |               30168 |            30168 | {"TabletMeta":48}                                          |
| LocalCatalog | InternalCatalogMemoryTracker |               33424 |            33424 | {"Partition":5}                                            |
+--------------+------------------------------+---------------------+------------------+------------------------------------------------------------+
```

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exec/schema_scanner/sys_fe_memory_usage.cpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#include "runtime/runtime_state.h"
+#include "runtime/string_value.h"
+
+namespace starrocks {
+
+SchemaScanner::ColumnDesc SysFeMemoryUsage::_s_columns[] = {
+        {"module_name", TYPE_VARCHAR, sizeof(StringValue), true},
+        {"class_name", TYPE_VARCHAR, sizeof(StringValue), true},
+        {"current_consumption", TYPE_VARCHAR, sizeof(StringValue), true},
+        {"peak_consumption", TYPE_VARCHAR, sizeof(StringValue), true}};
```
Comment (imay) [inline be/src/exec/schema_scanner/sys_fe_memory_usage.cpp:28]: It is better to use int64 as consumption, then we can use SQL expression to operate on this integer.
For better print, we can provide another function to make it human-friendly.

File: gensrc/thrift/FrontendService.thrift [hunk 2]
```diff
@@ -1616,6 +1616,21 @@ struct TFeLocksRes {
     1: optional list<TFeLocksItem> items
 }
 
+struct TFeMemoryItem {
+    1: optional string module_name
+    2: optional string class_name
+    3: optional string current_consumption
+    4: optional string peak_consumption
```
Comment (imay) [inline gensrc/thrift/FrontendService.thrift:1623]: use i64 instead

**Review by imay — COMMENTED** [review]
Add some sql-test to cover this new sys table

File: fe/fe-core/src/main/java/com/starrocks/catalog/system/sys/SysFeMemoryUsage.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        TFeMemoryRes response = new TFeMemoryRes();
+
+        MemoryUsageTracker.MEMORY_USAGE.forEach((moduleName, module) -> {
+            if (module != null) {
+                module.forEach((className, memoryStat) -> {
+                    TFeMemoryItem item = new TFeMemoryItem();
+                    item.setModule_name(moduleName);
+                    item.setClass_name(className);
+                    item.setCurrent_consumption(memoryStat.getCurrentConsumption());
+                    item.setPeak_consumption(memoryStat.getPeakConsumption());
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/catalog/system/sys/SysFeMemoryUsage.java:76]: can we also add object counters info as a json string and show it in the table?

**Review by nshangyiming — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by Astralidea** [comment]
https://github.com/Mergifyio backport branch-3.2

**Pull request merged**